pool.go

```diff
@@ -46,8 +38,35 @@ func (p *Pool) SetStrategy(strategy Strategy) error {
 	return nil
 }
 
+// Add .
+func (p *Pool) Add(ns Namespace) {
+	s := &ring.Ring{Value: ns}
+	r := p.Nodes.Link(s)
+	s.Link(r)
+}
+
+// Drop .
+func (p *Pool) Drop(domain string) error {
+	for i := 0; i < p.Nodes.Len(); i++ {
+		ns, ok := p.Nodes.Value.(Namespace)
+		if !ok {
+			return errors.New("Could not typecast Ring.Value to Namespace")
+		}
+
+		if ns.Domain == domain {
+			prev := p.Nodes.Prev()
+			next := p.Nodes.Next()
+			prev.Link(next)
+			p.Nodes = prev
```

Comment:
I'm not too familiar with the ring package. Do we have to set something to `nil` here for garbage collection?